File: en/java-jaxrs/src/main/java/com/example/controller/ProductController.java
package com.example.controller;

import jakarta.ws.rs.*;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;
import com.example.model.Product;
import com.example.dto.Result;
import com.example.dto.PageResult;
import java.math.BigDecimal;
import java.util.List;

@Path("/api/products")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class ProductController {
    
    @GET
    public Response listProducts(
            @QueryParam("keyword") String keyword,
            @QueryParam("categoryId") Long categoryId,
            @QueryParam("minPrice") BigDecimal minPrice,
            @QueryParam("maxPrice") BigDecimal maxPrice,
            @QueryParam("status") Integer status,
            @QueryParam("pageNum") @DefaultValue("1") Integer pageNum,
            @QueryParam("pageSize") @DefaultValue("10") Integer pageSize,
            @QueryParam("sortBy") @DefaultValue("id") String sortBy,
            @QueryParam("sortOrder") @DefaultValue("desc") String sortOrder) {
        PageResult<Product> result = new PageResult<>();
        return Response.ok(Result.success(result)).build();
    }
    
    @POST
    public Response createProduct(CreateProductReq req) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @GET
    @Path("/{id}")
    public Response getProduct(@PathParam("id") Long id) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @PUT
    @Path("/{id}")
    public Response updateProduct(@PathParam("id") Long id, UpdateProductReq req) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @DELETE
    @Path("/{id}")
    public Response deleteProduct(@PathParam("id") Long id) {
        return Response.ok(Result.success(null)).build();
    }
    
    @PUT
    @Path("/{id}/status")
    public Response updateStatus(@PathParam("id") Long id, @QueryParam("status") Integer status) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @PUT
    @Path("/{id}/price")
    public Response updatePrice(@PathParam("id") Long id, UpdatePriceReq req) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @PUT
    @Path("/{id}/stock")
    public Response updateStock(@PathParam("id") Long id, @QueryParam("stock") Integer stock) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @GET
    @Path("/{id}/detail")
    public Response getProductDetail(@PathParam("id") Long id) {
        ProductDetailVO detail = new ProductDetailVO();
        return Response.ok(Result.success(detail)).build();
    }
    
    @GET
    @Path("/category/{categoryId}")
    public Response getProductsByCategory(
            @PathParam("categoryId") Long categoryId,
            @QueryParam("pageNum") @DefaultValue("1") Integer pageNum,
            @QueryParam("pageSize") @DefaultValue("10") Integer pageSize) {
        PageResult<Product> result = new PageResult<>();
        return Response.ok(Result.success(result)).build();
    }
    
    @GET
    @Path("/hot")
    public Response getHotProducts(@QueryParam("limit") @DefaultValue("10") Integer limit) {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @GET
    @Path("/new")
    public Response getNewProducts(@QueryParam("limit") @DefaultValue("10") Integer limit) {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @GET
    @Path("/recommend")
    public Response getRecommendProducts(@QueryParam("limit") @DefaultValue("10") Integer limit) {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @POST
    @Path("/{id}/images")
    public Response uploadImages(@PathParam("id") Long id, List<String> imageUrls) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @DELETE
    @Path("/{id}/images")
    public Response deleteImages(@PathParam("id") Long id, @QueryParam("imageUrl") String imageUrl) {
        return Response.ok(Result.success(null)).build();
    }
    
    @POST
    @Path("/{id}/specs")
    public Response addSpec(@PathParam("id") Long id, CreateProductSpecReq req) {
        ProductSpecVO spec = new ProductSpecVO();
        return Response.ok(Result.success(spec)).build();
    }
    
    @PUT
    @Path("/{id}/specs/{specId}")
    public Response updateSpec(
            @PathParam("id") Long id,
            @PathParam("specId") Long specId,
            UpdateProductSpecReq req) {
        ProductSpecVO spec = new ProductSpecVO();
        return Response.ok(Result.success(spec)).build();
    }
    
    @DELETE
    @Path("/{id}/specs/{specId}")
    public Response deleteSpec(@PathParam("id") Long id, @PathParam("specId") Long specId) {
        return Response.ok(Result.success(null)).build();
    }
    
    @GET
    @Path("/{id}/specs")
    public Response getSpecs(@PathParam("id") Long id) {
        List<ProductSpecVO> specs = List.of();
        return Response.ok(Result.success(specs)).build();
    }
    
    @GET
    @Path("/{id}/reviews")
    public Response getReviews(
            @PathParam("id") Long id,
            @QueryParam("pageNum") @DefaultValue("1") Integer pageNum,
            @QueryParam("pageSize") @DefaultValue("10") Integer pageSize) {
        PageResult<Object> result = new PageResult<>();
        return Response.ok(Result.success(result)).build();
    }
    
    @GET
    @Path("/{id}/related")
    public Response getRelatedProducts(@PathParam("id") Long id, @QueryParam("limit") @DefaultValue("10") Integer limit) {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @POST
    @Path("/batch")
    public Response batchCreate(List<CreateProductReq> reqs) {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @DELETE
    @Path("/batch")
    public Response batchDelete(List<Long> ids) {
        return Response.ok(Result.success(null)).build();
    }
    
    @PUT
    @Path("/batch/status")
    public Response batchUpdateStatus(BatchUpdateStatusReq req) {
        return Response.ok(Result.success(null)).build();
    }
    
    @GET
    @Path("/export")
    public Response exportProducts(
            @QueryParam("categoryId") Long categoryId,
            @QueryParam("status") Integer status) {
        String fileUrl = "";
        return Response.ok(Result.success(fileUrl)).build();
    }
    
    @POST
    @Path("/import")
    public Response importProducts() {
        List<Product> products = List.of();
        return Response.ok(Result.success(products)).build();
    }
    
    @GET
    @Path("/search")
    public Response searchProducts(
            @QueryParam("keyword") String keyword,
            @QueryParam("pageNum") @DefaultValue("1") Integer pageNum,
            @QueryParam("pageSize") @DefaultValue("10") Integer pageSize) {
        PageResult<Product> result = new PageResult<>();
        return Response.ok(Result.success(result)).build();
    }
    
    @POST
    @Path("/{id}/view")
    public Response recordView(@PathParam("id") Long id) {
        return Response.ok(Result.success(null)).build();
    }
    
    @GET
    @Path("/{id}/statistics")
    public Response getStatistics(@PathParam("id") Long id) {
        ProductStatisticsVO statistics = new ProductStatisticsVO();
        return Response.ok(Result.success(statistics)).build();
    }
    
    @PUT
    @Path("/{id}/category")
    public Response updateCategory(@PathParam("id") Long id, @QueryParam("categoryId") Long categoryId) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @POST
    @Path("/{id}/tags")
    public Response addTags(@PathParam("id") Long id, List<String> tags) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @DELETE
    @Path("/{id}/tags")
    public Response removeTag(@PathParam("id") Long id, @QueryParam("tag") String tag) {
        Product product = new Product();
        return Response.ok(Result.success(product)).build();
    }
    
    @GET
    @Path("/{id}/tags")
    public Response getTags(@PathParam("id") Long id) {
        List<String> tags = List.of();
        return Response.ok(Result.success(tags)).build();
    }
}

class CreateProductReq {
    private String name;
    private String description;
    private BigDecimal price;
    private BigDecimal originalPrice;
    private Long categoryId;
    private Integer stock;
    private String mainImage;
    private String images;
    private String unit;
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public BigDecimal getOriginalPrice() { return originalPrice; }
    public void setOriginalPrice(BigDecimal originalPrice) { this.originalPrice = originalPrice; }
    public Long getCategoryId() { return categoryId; }
    public void setCategoryId(Long categoryId) { this.categoryId = categoryId; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
    public String getMainImage() { return mainImage; }
    public void setMainImage(String mainImage) { this.mainImage = mainImage; }
    public String getImages() { return images; }
    public void setImages(String images) { this.images = images; }
    public String getUnit() { return unit; }
    public void setUnit(String unit) { this.unit = unit; }
}

class UpdateProductReq {
    private String name;
    private String description;
    private BigDecimal price;
    private BigDecimal originalPrice;
    private Long categoryId;
    private String mainImage;
    private String images;
    private String unit;
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public BigDecimal getOriginalPrice() { return originalPrice; }
    public void setOriginalPrice(BigDecimal originalPrice) { this.originalPrice = originalPrice; }
    public Long getCategoryId() { return categoryId; }
    public void setCategoryId(Long categoryId) { this.categoryId = categoryId; }
    public String getMainImage() { return mainImage; }
    public void setMainImage(String mainImage) { this.mainImage = mainImage; }
    public String getImages() { return images; }
    public void setImages(String images) { this.images = images; }
    public String getUnit() { return unit; }
    public void setUnit(String unit) { this.unit = unit; }
}

class UpdatePriceReq {
    private BigDecimal price;
    private BigDecimal originalPrice;
    
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public BigDecimal getOriginalPrice() { return originalPrice; }
    public void setOriginalPrice(BigDecimal originalPrice) { this.originalPrice = originalPrice; }
}

class ProductDetailVO {
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private BigDecimal originalPrice;
    private Long categoryId;
    private String categoryName;
    private Integer stock;
    private Integer sales;
    private String mainImage;
    private List<String> images;
    private List<ProductSpecVO> specs;
    private Double averageRating;
    private Integer reviewCount;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public BigDecimal getOriginalPrice() { return originalPrice; }
    public void setOriginalPrice(BigDecimal originalPrice) { this.originalPrice = originalPrice; }
    public Long getCategoryId() { return categoryId; }
    public void setCategoryId(Long categoryId) { this.categoryId = categoryId; }
    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
    public Integer getSales() { return sales; }
    public void setSales(Integer sales) { this.sales = sales; }
    public String getMainImage() { return mainImage; }
    public void setMainImage(String mainImage) { this.mainImage = mainImage; }
    public List<String> getImages() { return images; }
    public void setImages(List<String> images) { this.images = images; }
    public List<ProductSpecVO> getSpecs() { return specs; }
    public void setSpecs(List<ProductSpecVO> specs) { this.specs = specs; }
    public Double getAverageRating() { return averageRating; }
    public void setAverageRating(Double averageRating) { this.averageRating = averageRating; }
    public Integer getReviewCount() { return reviewCount; }
    public void setReviewCount(Integer reviewCount) { this.reviewCount = reviewCount; }
}

class ProductSpecVO {
    private Long id;
    private String name;
    private String value;
    private BigDecimal price;
    private Integer stock;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
}

class CreateProductSpecReq {
    private String name;
    private String value;
    private BigDecimal price;
    private Integer stock;
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
}

class UpdateProductSpecReq {
    private String name;
    private String value;
    private BigDecimal price;
    private Integer stock;
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public Integer getStock() { return stock; }
    public void setStock(Integer stock) { this.stock = stock; }
}

class BatchUpdateStatusReq {
    private List<Long> ids;
    private Integer status;
    
    public List<Long> getIds() { return ids; }
    public void setIds(List<Long> ids) { this.ids = ids; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
}

class ProductStatisticsVO {
    private Long productId;
    private Integer views;
    private Integer sales;
    private Integer favorites;
    private Double averageRating;
    private Integer reviewCount;
    
    public Long getProductId() { return productId; }
    public void setProductId(Long productId) { this.productId = productId; }
    public Integer getViews() { return views; }
    public void setViews(Integer views) { this.views = views; }
    public Integer getSales() { return sales; }
    public void setSales(Integer sales) { this.sales = sales; }
    public Integer getFavorites() { return favorites; }
    public void setFavorites(Integer favorites) { this.favorites = favorites; }
    public Double getAverageRating() { return averageRating; }
    public void setAverageRating(Double averageRating) { this.averageRating = averageRating; }
    public Integer getReviewCount() { return reviewCount; }
    public void setReviewCount(Integer reviewCount) { this.reviewCount = reviewCount; }
}

File: cn/kotlin-springmvc/src/main/kotlin/com/example/controller/ProductController.kt
package com.example.controller

import com.example.dto.PageResult
import com.example.dto.Result
import com.example.model.Product
import org.springframework.web.bind.annotation.*
import java.math.BigDecimal

@RestController
@RequestMapping("/api/products")
class ProductController {
    
    @GetMapping
    fun listProducts(
        @RequestParam(required = false) keyword: String?,
        @RequestParam(required = false) categoryId: Long?,
        @RequestParam(required = false) minPrice: BigDecimal?,
        @RequestParam(required = false) maxPrice: BigDecimal?,
        @RequestParam(required = false) status: Int?,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int,
        @RequestParam(defaultValue = "id") sortBy: String,
        @RequestParam(defaultValue = "desc") sortOrder: String
    ): Result<PageResult<Product>> {
        val result = PageResult<Product>()
        return Result.success(result)
    }
    
    @PostMapping
    fun createProduct(@RequestBody req: CreateProductReq): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @GetMapping("/{id}")
    fun getProduct(@PathVariable id: Long): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @PutMapping("/{id}")
    fun updateProduct(@PathVariable id: Long, @RequestBody req: UpdateProductReq): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @DeleteMapping("/{id}")
    fun deleteProduct(@PathVariable id: Long): Result<Void> {
        return Result.success(null)
    }
    
    @PutMapping("/{id}/status")
    fun updateStatus(@PathVariable id: Long, @RequestParam status: Int): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @PutMapping("/{id}/price")
    fun updatePrice(@PathVariable id: Long, @RequestBody req: UpdatePriceReq): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @PutMapping("/{id}/stock")
    fun updateStock(@PathVariable id: Long, @RequestParam stock: Int): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @GetMapping("/{id}/detail")
    fun getProductDetail(@PathVariable id: Long): Result<ProductDetailVO> {
        val detail = ProductDetailVO()
        return Result.success(detail)
    }
    
    @GetMapping("/category/{categoryId}")
    fun getProductsByCategory(
        @PathVariable categoryId: Long,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Product>> {
        val result = PageResult<Product>()
        return Result.success(result)
    }
    
    @GetMapping("/hot")
    fun getHotProducts(@RequestParam(defaultValue = "10") limit: Int): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @GetMapping("/new")
    fun getNewProducts(@RequestParam(defaultValue = "10") limit: Int): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @GetMapping("/recommend")
    fun getRecommendProducts(@RequestParam(defaultValue = "10") limit: Int): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @PostMapping("/{id}/images")
    fun uploadImages(@PathVariable id: Long, @RequestBody imageUrls: List<String>): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @DeleteMapping("/{id}/images")
    fun deleteImages(@PathVariable id: Long, @RequestParam imageUrl: String): Result<Void> {
        return Result.success(null)
    }
    
    @PostMapping("/{id}/specs")
    fun addSpec(@PathVariable id: Long, @RequestBody req: CreateProductSpecReq): Result<ProductSpecVO> {
        val spec = ProductSpecVO()
        return Result.success(spec)
    }
    
    @PutMapping("/{id}/specs/{specId}")
    fun updateSpec(
        @PathVariable id: Long,
        @PathVariable specId: Long,
        @RequestBody req: UpdateProductSpecReq
    ): Result<ProductSpecVO> {
        val spec = ProductSpecVO()
        return Result.success(spec)
    }
    
    @DeleteMapping("/{id}/specs/{specId}")
    fun deleteSpec(@PathVariable id: Long, @PathVariable specId: Long): Result<Void> {
        return Result.success(null)
    }
    
    @GetMapping("/{id}/specs")
    fun getSpecs(@PathVariable id: Long): Result<List<ProductSpecVO>> {
        val specs = listOf<ProductSpecVO>()
        return Result.success(specs)
    }
    
    @GetMapping("/{id}/reviews")
    fun getReviews(
        @PathVariable id: Long,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Any>> {
        val result = PageResult<Any>()
        return Result.success(result)
    }
    
    @GetMapping("/{id}/related")
    fun getRelatedProducts(@PathVariable id: Long, @RequestParam(defaultValue = "10") limit: Int): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @PostMapping("/batch")
    fun batchCreate(@RequestBody reqs: List<CreateProductReq>): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @DeleteMapping("/batch")
    fun batchDelete(@RequestBody ids: List<Long>): Result<Void> {
        return Result.success(null)
    }
    
    @PutMapping("/batch/status")
    fun batchUpdateStatus(@RequestBody req: BatchUpdateStatusReq): Result<Void> {
        return Result.success(null)
    }
    
    @GetMapping("/export")
    fun exportProducts(
        @RequestParam(required = false) categoryId: Long?,
        @RequestParam(required = false) status: Int?
    ): Result<String> {
        val fileUrl = ""
        return Result.success(fileUrl)
    }
    
    @PostMapping("/import")
    fun importProducts(): Result<List<Product>> {
        val products = listOf<Product>()
        return Result.success(products)
    }
    
    @GetMapping("/search")
    fun searchProducts(
        @RequestParam keyword: String,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Product>> {
        val result = PageResult<Product>()
        return Result.success(result)
    }
    
    @PostMapping("/{id}/view")
    fun recordView(@PathVariable id: Long): Result<Void> {
        return Result.success(null)
    }
    
    @GetMapping("/{id}/statistics")
    fun getStatistics(@PathVariable id: Long): Result<ProductStatisticsVO> {
        val statistics = ProductStatisticsVO()
        return Result.success(statistics)
    }
    
    @PutMapping("/{id}/category")
    fun updateCategory(@PathVariable id: Long, @RequestParam categoryId: Long): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @PostMapping("/{id}/tags")
    fun addTags(@PathVariable id: Long, @RequestBody tags: List<String>): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @DeleteMapping("/{id}/tags")
    fun removeTag(@PathVariable id: Long, @RequestParam tag: String): Result<Product> {
        val product = Product()
        return Result.success(product)
    }
    
    @GetMapping("/{id}/tags")
    fun getTags(@PathVariable id: Long): Result<List<String>> {
        val tags = listOf<String>()
        return Result.success(tags)
    }
}

data class CreateProductReq(
    val name: String?,
    val description: String?,
    val price: BigDecimal?,
    val originalPrice: BigDecimal?,
    val categoryId: Long?,
    val stock: Int?,
    val mainImage: String?,
    val images: String?,
    val unit: String?
)

data class UpdateProductReq(
    val name: String?,
    val description: String?,
    val price: BigDecimal?,
    val originalPrice: BigDecimal?,
    val categoryId: Long?,
    val mainImage: String?,
    val images: String?,
    val unit: String?
)

data class UpdatePriceReq(
    val price: BigDecimal?,
    val originalPrice: BigDecimal?
)

data class ProductDetailVO(
    val id: Long? = null,
    val name: String? = null,
    val description: String? = null,
    val price: BigDecimal? = null,
    val originalPrice: BigDecimal? = null,
    val categoryId: Long? = null,
    val categoryName: String? = null,
    val stock: Int? = null,
    val sales: Int? = null,
    val mainImage: String? = null,
    val images: List<String>? = null,
    val specs: List<ProductSpecVO>? = null,
    val averageRating: Double? = null,
    val reviewCount: Int? = null
)

data class ProductSpecVO(
    val id: Long? = null,
    val name: String? = null,
    val value: String? = null,
    val price: BigDecimal? = null,
    val stock: Int? = null
)

data class CreateProductSpecReq(
    val name: String?,
    val value: String?,
    val price: BigDecimal?,
    val stock: Int?
)

data class UpdateProductSpecReq(
    val name: String?,
    val value: String?,
    val price: BigDecimal?,
    val stock: Int?
)

data class BatchUpdateStatusReq(
    val ids: List<Long>?,
    val status: Int?
)

data class ProductStatisticsVO(
    val productId: Long? = null,
    val views: Int? = null,
    val sales: Int? = null,
    val favorites: Int? = null,
    val averageRating: Double? = null,
    val reviewCount: Int? = null
)
